src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -225,25 +225,41 @@ private async Task<bool> CanUpdateGroupAccessAsync(ICollection<Collection> resou
 
     private async Task<bool> CanDeleteAsync(ICollection<Collection> resources, CurrentContextOrganization? org)
     {
-        // Owners, Admins, and users with DeleteAnyCollection permission can always delete collections
-        if (org is
-        { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
-        { Permissions.DeleteAnyCollection: true })
+        // Users with DeleteAnyCollection permission can always delete collections
+        if (org is { Permissions.DeleteAnyCollection: true })
         {
             return true;
         }
 
-        // Check for non-null org here: the user must be apart of the organization for this setting to take affect
-        // The limit collection management setting is disabled,
-        // ensure acting user has manage permissions for all collections being deleted
-        if (await GetOrganizationAbilityAsync(org) is { LimitCollectionCreationDeletion: false })
+        var organizationAbility = await GetOrganizationAbilityAsync(org);
+
+        // If AllowAdminAccessToAllCollectionItems is true, Owners and Admins can delete any collection, regardless of LimitCollectionCreationDeletion setting
+        if ((organizationAbility is { AllowAdminAccessToAllCollectionItems: true } ||
+             !_featureService.IsEnabled(FeatureFlagKeys.FlexibleCollectionsV1)) &&
+            org is { Type: OrganizationUserType.Owner or OrganizationUserType.Admin })
+        {
+            return true;
+        }
+
+        var canManageCollections = await CanManageCollectionsAsync(resources);
+
+        // LimitCollectionCreationDeletion is false, AllowAdminAccessToAllCollectionItems setting is irrelevant.
+        // Ensure acting user has manage permissions for all collections being deleted
+        if (organizationAbility is { LimitCollectionCreationDeletion: false })
         {
-            var canManageCollections = await CanManageCollectionsAsync(resources);
             if (canManageCollections)
             {
                 return true;
             }
         }
+        else
+        // LimitCollectionCreationDeletion is true, only Owners and Admins can delete collections they manage
+        {
+            if (org is { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } && canManageCollections)
+            {
+                return true;
+            }
+        }
```

Comment:
How you've written this works and I think this has turned out pretty well after much discussion about the business logic side of things.

One suggestion, I think this could be expressed more succinctly:

```cs
var canDeleteManagedCollections = organizationAbility is { LimitCollectionCreationDeletion: false } ||
  org is { Type: OrganizationUserType.Owner or OrganizationUserType.Admin };
if (canDeleteManagedCollections && await canManageCollections(resources))
{
  return true;
}
```

Reply:
Thank you!